### PR TITLE
Gentle TightConstraintSet kwarg

### PR DESCRIPTION
**Pull request by pgkirsch**
I'd like to have a kwarg that allows it to print a warning instead of throwing an error.

This is my proposed change.

### CODE REVIEW

**Comment by pgkirsch** [comment]
retest this please

**Comment by whoburg** [comment]
retest this please

File: gpkit/constraints/tight.py [hunk 1]
```diff
@@ -21,10 +23,13 @@ def process_result(self, result):
...
                 rel_diff = abs(1 - leftsubbed/rightsubbed)
                 if rel_diff >= self.reltol:
-                    raise ValueError("Tightness requirement not met"
-                                     " for constraint %s because %s"
-                                     " evaluated to %s but %s evaluated"
-                                     " to %s"
-                                     % (constraint,
-                                        constraint.left, leftsubbed,
-                                        constraint.right, rightsubbed))
+                    if self.raiseerror is True:
```
Comment (whoburg) [inline gpkit/constraints/tight.py]: this should be `if self.raiseerror:`.

**Comment by galbramc** [comment]
test this please

**Comment by galbramc** [comment]
test this please

**Comment by whoburg** [comment]
@pgkirsch I added a commit that tweaks things slightly -- are you happy with it? If so and if this passes tests, we can merge.

**Comment by pgkirsch** [comment]
Hmm I had used a different warning message because I see users perhaps wanting to know whether a constraint is tight but not really believing that the tightness is a requirement, so it seemed like a more gentle warning was appropriate. I like having just one `msg` though, but could we change it to something that doesn't mention a requirement being met?

**Comment by whoburg** [comment]
Oh sure -- added a commit aiming for that. My goal was to have the values appear in the warning, figured it might be useful someday.

**Comment by whoburg** [comment]
@pgkirsch, you happy with this?

**Comment by pgkirsch** [comment]
Awesome thanks. And yes I agree having values is helpful.

**Pull request merged**